Userland/Libraries/LibGL/Tex/NameAllocator.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGL/Tex/NameAllocator.h>
+
+namespace GL
+{
+
+void TextureNameAllocator::allocate(GLsizei count, GLuint *textures)
+{
+    for (auto i = 0; i < count; ++i) {
+        if (!m_free_texture_names.is_empty()) {
+            textures[i] = m_free_texture_names.top();
+            m_free_texture_names.pop();
+        } else {
+            // We're out of free previously allocated names. Let's allocate a new contiguous amount from the
+            // last known texture name
+            textures[i] = m_last_texture_id++;
```

Comment:
```suggestion
            textures[i] = ++m_last_texture_id;
```
The first valid name is 1, not 0. So either preincrement here or initialize m_last_texture_id with 0.

Reply:
Good spot! I've changed the initialised value of `m_last_texture_id` to `1`